Automattic/wp-calypso
# Add hot reloading

## Proposed Changes

* Try to add hot-reloading
  *  Noting that Sergio removed the `HotModuleReplacementPlugin()` in #50187 because the hot updates were failing - and he mentioned we should try to get it working later.
  * I tried various things and was eventually able to get something working with `pmmmwh/react-refresh-webpack-plugin`.

* PROS:
  *  It seems to work :)
* CONS:
  * I don't really know what I'm doing and just fiddled with stuff until it worked.
  * I really know very little about calypso's complicated setup.
  * Not tested very much.
  * Would not be surprised if redux, wordpress/data etc break it.
  * Uses pre 1.0 plugin https://github.com/pmmmwh/react-refresh-webpack-plugin to take advantage of react's new "fast refresh"
  * Had to `yarn add -D` packages (but hopefully they don't affect production build?)

## Testing Instructions

* `yarn && yarn start`
* Use calypso
* Make changes in your editor and see if they hot reload


https://user-images.githubusercontent.com/937354/233733014-568c6c4a-2c9c-4cc3-be8a-bad3ce1d612e.mp4



## Pre-merge Checklist
